src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -298,13 +298,30 @@ private void RemoveEntry(NodeSearchElement entry)
             var treeStack = new Stack<NodeCategoryViewModel>();
             var nameStack = new Stack<string>(entry.Categories.Reverse());
             var target = libraryRoot;
+            bool isCheckedForClassesCategory = false;
             while (nameStack.Any())
             {
                 var next = nameStack.Pop();
                 var categories = target.SubCategories;
                 var newTarget = categories.FirstOrDefault(c => c.Name == next);
-                if (newTarget == default(NodeCategoryViewModel))
+                if (newTarget == null)
                 {
+                    // Latest item in categories list can be a class.
+                    // All classes are situated in classes container.
+                    // So nameStack should be populated with ClassesDefaultName
+                    // and class name. And checked again.
+                    // For example: "MyAssembly.MyNamespace.ClassCandidate" not found.
+                    // Should be checked for "MyAssembly.MyNamespace.Classes.ClassCandidate"
+                    //
+                    if (!isCheckedForClassesCategory && nameStack.Count == 0)
+                    {
+                        nameStack.Push(next);
+                        nameStack.Push(Configurations.ClassesDefaultName);
+
+                        isCheckedForClassesCategory = true;
+                        continue;
+                    }
+
```

Comment:
Thanks a lot for the rework, @vmoyseenko, this looks much better than the one in `master`! Sorry I have to be a little nit-picky on things, but a properly API design should never return `null` for a collection (which means the caller should never have to check for `null`). If this method call fails, please return a `new List<NodeCategoryViewModel>()` instead (because `nodesOnBranch` may be non-empty).